liota/entities/devices/sensor_tag.py

```diff
@@ -0,0 +1,451 @@
+#                                                                             #
+#  - Redistributions in binary form must reproduce the above copyright        #
+#      notice, this list of conditions and the following disclaimer in the    #
+#      documentation and/or other materials provided with the distribution.   #
+#                                                                             #
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"#
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE  #
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE #
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE  #
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR        #
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+import time
+import Queue
+import logging
+from threading import Thread
+
+from aenum import UniqueEnum
+from bluepy.sensortag import SensorTag
+from bluepy.btle import BTLEException
+
+from liota.entities.devices.device import Device
+from liota.lib.utilities.utility import systemUUID
+
+log = logging.getLogger(__name__)
+
+
+class Sensors(UniqueEnum):
+    """
+    Enum for sensors that can be enabled in a SensorTag Device.
+
+    Note: Use ALL to enable all the sensors.
+    """
+    TEMPERATURE = 0
+    HUMIDITY = 1
+    BAROMETER = 2
+    ACCELEROMETER = 3
+    MAGNETOMETER = 4
+    GYROSCOPE = 5
+    LIGHTMETER = 6
+    BATTERY_LEVEL = 7
+    # All the above sensors
+    ALL = 8
+
+
+class SensorTagDevice(Device, SensorTag):
+    """
+    SensorTag Device Entity.
+
+    Texas Instrument's SensorTag
+    http://www.ti.com/ww/en/wireless_connectivity/sensortag2015/?INTC=SensorTag&HQS=sensortag
+    """
+
+    def __init__(self, name, device_mac, entity_type="Device"):
+        Device.__init__(
+            self,
+            name=name,
+            entity_type=entity_type,
+            entity_id=systemUUID().get_uuid(name)
+        )
+        SensorTag.__init__(
+            self,
+            addr=device_mac
+        )
+
+
+class SensorTagCollector(Thread):
+    """
+    Helper-Thread to connect with SensorTagDevice over BLE (Bluetooth Low Energy)
+
+    Facilitates collecting sensor metrics even in-case of dis-connectivity due to external factors, without any need
+    of manual restart of the program.
+
+    Use this class to collect metrics from SensorTag.
```

Comment:
There shouldn't be a separate class for collecting metrics, make it part of SensorTagDevice class. Creates confusion and complexity for user.